core/email.go

```diff
@@ -24,12 +33,39 @@ func (email emailAddress) String() string {
 // This uses RFC-5322 to determine valid e-mail addresses, e.g. "Biggie Smalls <notorious@example.com>"
 func NewEmailAddress(address string) (EmailAddress, error) {
 	if len(address) == 0 {
-		return nil, errors.New("e-mail address cannot be empty")
+		return nil, errors.Join(ErrInvalidEmailAddress, ErrEmailAddressEmpty)
 	}
 	address = strings.ToLower(address)
 	_, err := mail.ParseAddress(address)
 	if err != nil {
-		return nil, fmt.Errorf("cannot parse e-mail address %q: %w", address, err)
+		return nil, errors.Join(
+			ErrInvalidEmailAddress,
+			fmt.Errorf("cannot parse e-mail address %q: %w", address, err),
+		)
 	}
 	return emailAddress{address}, nil
 }
+
+type EmailService interface {
+	// SendEmail will build and send a basic e-mail message containing both a HTML template version
+	// (optional) as well as a plaintext alternative (required).
+	SendEmail(
+		ctx context.Context,
+		address EmailAddress,
+		subject string,
+		template *templ.Component,
```

Comment:
moeten we het gebruik hiervan evt ergens documenteren? Het lijkt hier dat je dan gewoon bv `@ScheduleDetails(....)` meegeeft als input, is dat correct?

Reply:
Is het type niet al documentatie genoeg? 😛 ik kan altijd wel een voorbeeld toevoegen ofzo?
Dit is Go code en niet templ, dus bv `templates.MailWelcome` ofzo. Je kan op deze manier wel geen extra argumenten meegeven aan die template (behalve ctx die er altijd is), dus mogelijks willen we daar wel nog een iets andere syntax zodat je bv `templates.MailWelcome(user)` kan doen

Reply:
Ah ja, ik vroeg het idd in verband met mogelijk argumenten. In python zou ik hier losse args/kwargs doen die je dan doorgeeft aan de template :-)